Mathlib/CategoryTheory/Limits/Sifted.lean

```diff
@@ -0,0 +1,386 @@
+/-
+Copyright (c) 2024 Robin Carlier. All rights reserved.
+Released under Apache 2.0 license as described in the file LICENSE.
+Authors: Robin Carlier
+-/
+import Mathlib.CategoryTheory.Limits.Fubini
+import Mathlib.CategoryTheory.Monoidal.Types.Basic
+import Mathlib.CategoryTheory.Closed.Types
+import Mathlib.CategoryTheory.Closed.FunctorToTypes
+import Mathlib.CategoryTheory.ChosenFiniteProducts.FunctorCategory
+import Mathlib.CategoryTheory.Limits.IsConnected
+import Mathlib.CategoryTheory.Limits.FilteredColimitCommutesFiniteLimit
+/-!
+# Sifted categories
+
+A category `C` is Sifted if `C` is nonempty and the diagonal functor `C ⥤ C × C` is final.
+Sifted categories can be caracterized as those such that the colimit functor `(C ⥤ Type) ⥤ Type `
+preserves finite products. We achieve this characterization in this file, as well as providing some
+API to produce `IsSifted` instances.
+
+## Main results
+- `colimPreservesFiniteProductsOfIsSifted`: The `Type`-valued colimit functor for sifted diagrams
+  preserves finite products.
+- `IsSiftedOfColimitPreservesFiniteProducts`: The converse: if the `Type`-valued colimit functor
+  preserves finite producs, the category is sifted.
+- `IsSiftedOfFinalFunctorFromSifted`: A category admitting a final functor from a sifted category is
+  itself sifted.
+- `IsSiftedOfIsFiltered`: A filtered category is sifted.
+- `IsSiftedOfHasBinaryCoproductsAndNonempty`: A nonempty category with binary copreducts is sifted.
+-/
+noncomputable section
+
+universe w v v₁ u u₁
+open CategoryTheory.MonoidalCategory CategoryTheory.Functor CategoryTheory.Limits CategoryTheory
+
+namespace CategoryTheory
+
+section ArbitraryUniverses
+
+variable (C : Type u) [Category.{v} C]
+
+/-- A category `C` `IsSiftedOrEmpty` if the diagonal functor `C ⥤ C × C` is final. -/
+class IsSiftedOrEmpty : Prop where
+  out : Final (diag C)
```

Comment:
I would think that `IsSiftedOrEmpty` could be abbreviation for `Final (diag C)`.